src/composables/fungibleTokens.ts

```diff
@@ -0,0 +1,288 @@
+import AedexV2PairACI from '@/lib/contracts/AedexV2PairACI.json';
+import ZeitTokenACI from '@/lib/contracts/FungibleTokenFullACI.json';
+import { AE_SYMBOL } from '@/protocols/aeternity/config';
+import { useAccounts } from './accounts';
+import { useAeSdk } from './aeSdk';
+import { useMiddleware } from './middleware';
+import { useTippingContracts } from './tippingContracts';
+import { useNetworks } from './networks';
+
+const availableTokens = ref<ITokenList>({});
+const tokensBalance = ref<Record<string, IToken[]>>({});
+
+export function useFungibleTokens() {
+  const store = useStore();
+  const { activeNetwork } = useNetworks();
+  const { getAeSdk } = useAeSdk({ store });
+  const { fetchFromMiddleware } = useMiddleware();
+  const { tippingContractAddresses } = useTippingContracts({ store });
+  const {
+    isLoggedIn,
+    accounts,
+    aeNextAccountIdx,
+    getLastActiveProtocolAccount,
+  } = useAccounts({ store });
+
+  function getTokenBalance(address?: string) {
+    const account = getLastActiveProtocolAccount(PROTOCOL_AETERNITY);
+    return tokensBalance.value[address || account?.address!] || [];
+  }
+
+  async function loadAvailableTokens() {
+    const response: IToken[] = camelCaseKeysDeep(await fetchAllPages(
+      () => fetchFromMiddleware('/v2/aex9?by=name&limit=100&direction=forward') as any,
+      fetchFromMiddleware as any,
+    ));
+
+    if (!response.length) {
+      availableTokens.value = {};
+    }
+
+    availableTokens.value = response.reduce((accumulator, token) => {
+      const { contractId, ...other } = token;
+
+      return {
+        ...accumulator,
+        [contractId]: { contractId, ...other },
+      };
+    }, {});
+  }
+
+  async function loadTokenBalances() {
+    if (!isLoggedIn.value) {
+      return;
+    }
+    const addresses = (accounts.value)
+      .filter((account) => account.protocol === PROTOCOL_AETERNITY)
+      .map((account) => account.address);
+
+    await Promise.all(addresses.map(async (address) => {
+      try {
+        const tokens: IToken[] = camelCaseKeysDeep(await fetchAllPages(
+          () => fetchFromMiddleware(`/v2/aex9/account-balances/${address}?limit=100`) as any,
+          fetchFromMiddleware as any,
+        ));
+        if (!tokens.length) {
+          return;
+        }
+
+        const balances = (tokens)
+          .filter((token) => availableTokens.value[token.contractId])
+          .map((token) => {
+            const availableToken = availableTokens.value[token.contractId];
+            const balance = toShiftedBigNumber(token.amount!, -availableToken.decimals);
+            const convertedBalance = Number(balance.toFixed(2));
+
+            return {
+              ...availableToken,
+              value: token.contractId,
+              text: `${convertedBalance} ${availableToken.symbol}`,
+              contractId: token.contractId,
+              balance,
+              convertedBalance,
+            } as IToken;
```

Comment:
If you use `as Something` the TS engine won't fully check the data. I suggest to first create a typed constant and then return it:
```
const token: IToken = { ... };
return token;
```